### ISSUE TITLE
Backup the list of installed addons

### ISSUE THREAD

**Issue by stdedos** [robweber/xbmcbackup#185]
Until Full backup is done (and stored, and updated automatically every now and then ... i.e. #173), is there some in-between ground for backing up the user-installed addons (i.e. not the dependencies)?
And if so ... is there also a way to "inject them" to Kodi?

I am imagining a post-clean-install restore, which, magically, writes the list of the addons that "should be installed" and closes Kodi. Then, I imagine, when Kodi boots up again, it will read and try to satisfy that list of "theoretically present" addons.

"If everything is done as they are supposed to", you don't need to carry the extension artifacts anymore, you get to restore your settings, and not waste space (carrying around artifacts) or time (manually trying to re-install the correct addons because you don't have a full backup i.e. by reading old logs that may be laying around).

Worst case, you can manually search/restore the addons one by one, since you know what you are looking for.

**Comment by robweber** [comment]
I think I follow what you're saying. This would be more of a list of addons that are in the userdata folder, just as like a txt file or something. You could then, in theory, tell Kodi just to redownload them instead of having to save the actual files and dependencies somewhere. Is that kind of what you mean? 

I'd have to research of Kodi has a mechanism to tell it to install an addon based on it's ID or something via Python or JSON-RPC.

**Comment by stdedos** [comment]
Yeah, I think you have the basic idea of what I am thinking about to the fullest.

Of course, concept / implementation is up to the availability and limitations. 😀
